diff --git a/cmake/onnxruntime_python.cmake b/cmake/onnxruntime_python.cmake
index 2788d326b..0d37892c3 100644
--- a/cmake/onnxruntime_python.cmake
+++ b/cmake/onnxruntime_python.cmake
@@ -438,9 +438,11 @@ endif()
 
 if (onnxruntime_BUILD_UNIT_TESTS)
   file(GLOB onnxruntime_python_test_srcs CONFIGURE_DEPENDS
-      "${ONNXRUNTIME_ROOT}/test/python/*.py"
-      "${ORTTRAINING_SOURCE_DIR}/test/python/*.py"
-      "${ORTTRAINING_SOURCE_DIR}/test/python/*.json"
+    "${ONNXRUNTIME_ROOT}/test/python/*.py"
+  )
+  file(GLOB onnxruntime_ottr_python_test_srcs CONFIGURE_DEPENDS
+    "${ORTTRAINING_SOURCE_DIR}/test/python/*.py"
+    "${ORTTRAINING_SOURCE_DIR}/test/python/*.json"
   )
   file(GLOB onnxruntime_python_quantization_test_srcs CONFIGURE_DEPENDS
       "${ONNXRUNTIME_ROOT}/test/python/quantization/*.py"
@@ -677,6 +679,9 @@ if (onnxruntime_BUILD_UNIT_TESTS)
     COMMAND ${CMAKE_COMMAND} -E copy
         ${onnxruntime_python_test_srcs}
         $<TARGET_FILE_DIR:${build_output_target}>
+        COMMAND ${CMAKE_COMMAND} -E copy
+        ${onnxruntime_ottr_python_test_srcs}
+        $<TARGET_FILE_DIR:${build_output_target}>
     COMMAND ${CMAKE_COMMAND} -E copy
         ${onnxruntime_python_quantization_test_srcs}
         $<TARGET_FILE_DIR:${build_output_target}>/quantization/
